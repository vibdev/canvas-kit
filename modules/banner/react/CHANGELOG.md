# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.25](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.24...@workday/canvas-kit-react-banner@0.1.25) (2019-06-20)


### Bug Fixes

* broad stroke accessibility fixes ([9803412](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/commits/9803412))





## [0.1.24](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.23...@workday/canvas-kit-react-banner@0.1.24) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.23](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.22...@workday/canvas-kit-react-banner@0.1.23) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.22](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.21...@workday/canvas-kit-react-banner@0.1.22) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.20...@workday/canvas-kit-react-banner@0.1.21) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.19...@workday/canvas-kit-react-banner@0.1.20) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.18...@workday/canvas-kit-react-banner@0.1.19) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.17...@workday/canvas-kit-react-banner@0.1.18) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.16...@workday/canvas-kit-react-banner@0.1.17) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.15...@workday/canvas-kit-react-banner@0.1.16) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.14...@workday/canvas-kit-react-banner@0.1.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.13...@workday/canvas-kit-react-banner@0.1.14) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.12...@workday/canvas-kit-react-banner@0.1.13) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.11...@workday/canvas-kit-react-banner@0.1.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.10...@workday/canvas-kit-react-banner@0.1.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.9...@workday/canvas-kit-react-banner@0.1.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.8...@workday/canvas-kit-react-banner@0.1.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.7...@workday/canvas-kit-react-banner@0.1.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.6...@workday/canvas-kit-react-banner@0.1.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.5...@workday/canvas-kit-react-banner@0.1.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.4...@workday/canvas-kit-react-banner@0.1.5) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.3...@workday/canvas-kit-react-banner@0.1.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.2...@workday/canvas-kit-react-banner@0.1.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.1...@workday/canvas-kit-react-banner@0.1.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## 0.1.1 (2019-04-30)

**Note:** Version bump only for package @workday/canvas-kit-react-banner
