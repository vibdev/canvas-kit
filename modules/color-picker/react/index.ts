import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
