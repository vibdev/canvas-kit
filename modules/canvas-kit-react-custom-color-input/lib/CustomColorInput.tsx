import * as React from 'react';
import styled from 'react-emotion';
import {focusRing} from '@workday/canvas-kit-react-common';
import {colors} from '@workday/canvas-kit-react-core';
import {css} from 'emotion';
import {IconButton} from '@workday/canvas-kit-react-button';
import {checkIcon} from '@workday/canvas-system-icons-web';

interface CustomColorInputState {
  /**
   * The hex value entered by the user
   */
  colorHexValue: string;
}

interface CustomColorInputProps {
  onSubmit: (color: string) => void;
}

const CustomHexInput = styled('input')({
  margin: 0,
  height: 40,
  WebkitAppearance: 'none',
  MozAppearance: 'none',
  borderColor: 'transparent',
  borderRadius: '2px',
  border: `1px solid ${colors.frenchVanilla400}`,
  boxSizing: 'border-box',
  textAlign: 'left',
  paddingLeft: '36px',
  marginRight: 8,
  '&:focus, &:active': {
    outline: 'none',
    ...focusRing(2, 0),
  },
});

const container = css({
  display: 'flex',
  boxSizing: 'border-box',
  position: 'relative',
});

const SwatchTile = styled('div')({
  position: 'absolute',
  cursor: 'pointer',
  height: '20px',
  width: '20px',
  top: '0',
  bottom: 0,
  margin: 'auto',
  left: 8,
  border: '1px solid grey',
  pointerEvents: 'none',
  borderRadius: '3px',
});

const selectedCustomHex = css({
  ...focusRing(2, 2),
});

export default class CustomColorInput extends React.Component<
  CustomColorInputProps,
  CustomColorInputState
> {
  private inputRef: React.RefObject<HTMLInputElement> = React.createRef();
  public constructor(props: CustomColorInputProps) {
    super(props);
    this.state = {
      colorHexValue: '',
    };
  }
  public render() {
    return (
      <div className={container}>
        <CustomHexInput
          onKeyPress={this.onKeyPress}
          placeholder="eg. #FFFFFF"
          innerRef={this.inputRef}
          onChange={this.onCustomHexChange}
          type="text"
          defaultValue={this.state.colorHexValue}
        />
        <SwatchTile
          style={{backgroundColor: `${this.state.colorHexValue || ''}`}}
          role="button"
          className={
            this.isValidHexValue(this.state.colorHexValue) && this.state.colorHexValue
              ? css`
                  ${selectedCustomHex};
                `
              : ''
          }
        />
        <IconButton
          onClick={this.handleSubmit}
          buttonType={IconButton.Types.Filled}
          icon={checkIcon}
        />
      </div>
    );
  }

  /**
   * If the hex code is valid, user can press enter to submit their custom hex color
   */
  private onKeyPress = (evt: React.KeyboardEvent<HTMLInputElement>) => {
    const isValidHex: boolean = this.isValidHexValue(this.state.colorHexValue);
    if (evt.key === 'Enter' && isValidHex) {
      this.setState({
        colorHexValue: this.state.colorHexValue,
      });
      this.props.onSubmit(this.state.colorHexValue);
    }
  };

  /**
   * On change, it checks if the hex code is valid and updates the custom hex swatch
   */
  private onCustomHexChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
    const isValidHex: boolean = this.isValidHexValue(evt.target.value);
    const newColorHexValue: string = isValidHex ? this.addPoundSign(evt.target.value) : '';
    this.setState({
      colorHexValue: newColorHexValue,
    });
  };

  /**
   * Checks if the hex code is valid and the right length
   */
  private isValidHexValue = (hexCode: string): boolean => {
    return /(^#?[0-9A-F]{6}$)|(^#?[0-9A-F]{3}$)/i.test(hexCode);
  };

  /**
   * Adds pound sign to the hex code if necessary
   */
  private addPoundSign = (hexCode: string): string => {
    return hexCode.slice(0, 1) === '#' ? hexCode : `#${hexCode}`;
  };

  private handleSubmit = (): void => {
    console.warn('input submit');
    // this.setState({
    //   submitedColorValue: this.state.colorHexValue,
    // });
    this.props.onSubmit(this.state.colorHexValue);
  };
}
