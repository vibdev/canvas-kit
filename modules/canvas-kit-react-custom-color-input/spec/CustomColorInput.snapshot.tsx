import * as React from 'react';
import CustomColorInput from '../lib/CustomColorInput';
import * as renderer from 'react-test-renderer';

describe('CustomColorInput Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <CustomColorInput selectedHexColor={''} onSubmit={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with a selected color', () => {
    const component = renderer.create(
      <CustomColorInput selectedHexColor={'#e6e'} onSubmit={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with a selected color without #', () => {
    const component = renderer.create(
      <CustomColorInput selectedHexColor={'DD5'} onSubmit={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
