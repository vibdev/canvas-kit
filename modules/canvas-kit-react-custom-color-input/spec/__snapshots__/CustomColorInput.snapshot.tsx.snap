// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomColorInput Snapshots renders as expected 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  position: relative;
}

.emotion-0 {
  margin: 0;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #bdbdbd;
  box-sizing: border-box;
  text-align: left;
  padding-left: 36px;
  margin-right: 8px;
}

.emotion-0:focus,
.emotion-0:active {
  border-color: transparent;
  outline: none;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-1 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0;
  bottom: 0;
  left: 8px;
  margin: auto;
  border: 1px solid #b9c0c7;
  pointer-events: none;
  border-radius: 2px;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: #f0f1f2;
  width: 40px;
  height: 40px;
}

.emotion-3:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-3:disabled,
.emotion-3:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-3 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-3 span .wd-icon-fill,
.emotion-3 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-3:focus span .wd-icon-fill,
.emotion-3:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-3:hover:focus {
  background-color: #dfe2e6;
}

.emotion-3:active,
.emotion-3:focus:active,
.emotion-3:hover:active {
  background-color: #ced3d9;
}

.emotion-3:hover {
  background-color: #dfe2e6;
}

.emotion-3:hover span .wd-icon-fill,
.emotion-3:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-3:disabled,
.emotion-3:active:disabled,
.emotion-3:focus:disabled,
.emotion-3:hover:disabled {
  background-color: #f6f7f8;
}

.emotion-3:disabled span .wd-icon-fill,
.emotion-3:active:disabled span .wd-icon-fill,
.emotion-3:focus:disabled span .wd-icon-fill,
.emotion-3:hover:disabled span .wd-icon-fill,
.emotion-3:disabled span .wd-icon-accent,
.emotion-3:active:disabled span .wd-icon-accent,
.emotion-3:focus:disabled span .wd-icon-accent,
.emotion-3:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-3:not([disabled]):focus,
.emotion-3:not([disabled]):active {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-4"
>
  <input
    className="emotion-0"
    defaultValue=""
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="eg. #FFFFFF"
    type="text"
  />
  <div
    className="emotion-1"
    role="button"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <button
    className="emotion-3"
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </button>
</div>
`;

exports[`CustomColorInput Snapshots renders with a selected color 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  position: relative;
}

.emotion-0 {
  margin: 0;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #bdbdbd;
  box-sizing: border-box;
  text-align: left;
  padding-left: 36px;
  margin-right: 8px;
}

.emotion-0:focus,
.emotion-0:active {
  border-color: transparent;
  outline: none;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: #f0f1f2;
  width: 40px;
  height: 40px;
}

.emotion-3:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-3:disabled,
.emotion-3:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-3 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-3 span .wd-icon-fill,
.emotion-3 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-3:focus span .wd-icon-fill,
.emotion-3:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-3:hover:focus {
  background-color: #dfe2e6;
}

.emotion-3:active,
.emotion-3:focus:active,
.emotion-3:hover:active {
  background-color: #ced3d9;
}

.emotion-3:hover {
  background-color: #dfe2e6;
}

.emotion-3:hover span .wd-icon-fill,
.emotion-3:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-3:disabled,
.emotion-3:active:disabled,
.emotion-3:focus:disabled,
.emotion-3:hover:disabled {
  background-color: #f6f7f8;
}

.emotion-3:disabled span .wd-icon-fill,
.emotion-3:active:disabled span .wd-icon-fill,
.emotion-3:focus:disabled span .wd-icon-fill,
.emotion-3:hover:disabled span .wd-icon-fill,
.emotion-3:disabled span .wd-icon-accent,
.emotion-3:active:disabled span .wd-icon-accent,
.emotion-3:focus:disabled span .wd-icon-accent,
.emotion-3:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-3:not([disabled]):focus,
.emotion-3:not([disabled]):active {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0;
  bottom: 0;
  left: 8px;
  margin: auto;
  border: 1px solid #b9c0c7;
  pointer-events: none;
  border-radius: 2px;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-4 {
  display: inline-block;
  position: absolute;
  left: 8px;
  top: 8px;
}

.emotion-4 svg {
  display: block;
}

.emotion-4 .wd-icon-fill {
  fill: #fff;
}

.emotion-4:hover .wd-icon-fill {
  fill: #fff;
}

.emotion-4 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-4:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-4 .wd-icon-background {
  fill: transparent;
}

.emotion-4:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-5"
>
  <input
    className="emotion-0"
    defaultValue=""
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="eg. #FFFFFF"
    type="text"
  />
  <div
    className="emotion-1"
    role="button"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <button
    className="emotion-3"
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </button>
  <span
    className="emotion-4"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
  />
</div>
`;

exports[`CustomColorInput Snapshots renders with a selected color without # 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  position: relative;
}

.emotion-0 {
  margin: 0;
  height: 40px;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-color: transparent;
  border-radius: 4px;
  border: 1px solid #bdbdbd;
  box-sizing: border-box;
  text-align: left;
  padding-left: 36px;
  margin-right: 8px;
}

.emotion-0:focus,
.emotion-0:active {
  border-color: transparent;
  outline: none;
  -webkit-animation: animation-bvpr5k 100ms;
  animation: animation-bvpr5k 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #005cb9;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: #f0f1f2;
  width: 40px;
  height: 40px;
}

.emotion-3:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-3:disabled,
.emotion-3:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-3 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-3 span .wd-icon-fill,
.emotion-3 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-3:focus span .wd-icon-fill,
.emotion-3:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-3:hover:focus {
  background-color: #dfe2e6;
}

.emotion-3:active,
.emotion-3:focus:active,
.emotion-3:hover:active {
  background-color: #ced3d9;
}

.emotion-3:hover {
  background-color: #dfe2e6;
}

.emotion-3:hover span .wd-icon-fill,
.emotion-3:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-3:disabled,
.emotion-3:active:disabled,
.emotion-3:focus:disabled,
.emotion-3:hover:disabled {
  background-color: #f6f7f8;
}

.emotion-3:disabled span .wd-icon-fill,
.emotion-3:active:disabled span .wd-icon-fill,
.emotion-3:focus:disabled span .wd-icon-fill,
.emotion-3:hover:disabled span .wd-icon-fill,
.emotion-3:disabled span .wd-icon-accent,
.emotion-3:active:disabled span .wd-icon-accent,
.emotion-3:focus:disabled span .wd-icon-accent,
.emotion-3:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-3:not([disabled]):focus,
.emotion-3:not([disabled]):active {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0;
  bottom: 0;
  left: 8px;
  margin: auto;
  border: 1px solid #b9c0c7;
  pointer-events: none;
  border-radius: 2px;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-4 {
  display: inline-block;
  position: absolute;
  left: 8px;
  top: 8px;
}

.emotion-4 svg {
  display: block;
}

.emotion-4 .wd-icon-fill {
  fill: #fff;
}

.emotion-4:hover .wd-icon-fill {
  fill: #fff;
}

.emotion-4 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-4:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-4 .wd-icon-background {
  fill: transparent;
}

.emotion-4:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-5"
>
  <input
    className="emotion-0"
    defaultValue=""
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="eg. #FFFFFF"
    type="text"
  />
  <div
    className="emotion-1"
    role="button"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <button
    className="emotion-3"
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </button>
  <span
    className="emotion-4"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
  />
</div>
`;
